tests/unit/graphql/merge.spec.js

```diff
@@ -0,0 +1,292 @@
+          const tests = [
+            {
+              a: true,
+              b: false
+            },
+            {
+              a: 'test',
+              b: 'production'
+            },
+            {
+              a: 1,
+              b: 10
+            },
+            {
+              a: 1.0,
+              b: 10.0
+            }
+          ]
+          for (const test of tests) {
+            const queryA = gql`query {
+              jobs (sort: ${test.a}) { id }
+            }`
+            const queryB = gql`query {
+              jobs (sort: ${test.b}) { id }
+            }`
+            expect(() => mergeQueries(queryA, queryB)).to.throw()
+          }
+        })
+        it('should throw an error if using different objects', () => {
+          const queryA = gql`query {
+            jobs (sort: { keys: ["a"] }) {
+              id
+            }
+          }`
+          const queryB = gql`query {
+            jobs (sort: { array: ["a"] }) {
+              id
+            }
+          }`
+          expect(() => mergeQueries(queryA, queryB)).to.throw()
+        })
+      })
+      describe('Values', () => {
+      })
+      it('should merge arguments', () => {
+        const queryA = gql`query A {
+          jobs (sort: true, name: "a", keys: ["a", "b"], expired: 100, flag: $flag, test: null) { id }
+        }`
+        const queryB = gql`query B {
+          jobs (keys: ["c"], age: 10, expired: 100, flag: $flag, test: null) { name }
+        }`
+        const merged = mergeQueries(queryA, queryB)
+        const expected = gql`query A {
+          jobs (sort: true, name: "a", keys: ["a", "b", "c"], expired: 100, flag: $flag, test: null, age: 10) { id name }
+        }`
+        assertQueriesAreEqual(merged, expected)
+      })
+      it('should merge arguments with objects', () => {
+        // Objects tests are a bit longer, so breaking into two basic tests
+        const queryA = gql`query A {
+          jobs (user: { id: 1, name: "cylc" }) {
+            id
+          }
+        }`
+        const queryB = gql`query B {
+          jobs (user: { id: 1, name: "cylc" }) {
+            name
+          }
+        }`
+        const merged = mergeQueries(queryA, queryB)
+        const expected = gql`query A {
+          jobs (user: { id: 1, name: "cylc" }) {
+            id
+            name
+          }
+        }`
+        assertQueriesAreEqual(merged, expected)
+      })
+    })
+  })
+  describe('Merge fragments', () => {
+    it('should merge two fragments', () => {
+      const queryA = gql`query A {
+        ...Jobs
+      }
+      
+      fragment Jobs on Job {
+        id
+      }
+      `
+      const queryB = gql`query B {
+        ...Jobs
+        ...Tasks
+      }
+      
+      fragment Jobs on Job {
+        id
+        name
+        status
+      }
+      
+      fragment Tasks on Task {
+        id
+        name
+        status
+      }
+      `
```

Comment:
Worth testing that merging works both ways i.e `{ 1, 2 } + { 2, 3 } = { 1, 2, 3 }`.

```suggestion
      const queryA = gql`query A {
        ...Jobs
      }
      
      fragment Jobs on Job {
        id
        name
      }
      `
      const queryB = gql`query B {
        ...Jobs
        ...Tasks
      }
      
      fragment Jobs on Job {
        name
        status
      }
      
      fragment Tasks on Task {
        id
        name
        status
      }
      `
```